sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -184,4 +186,39 @@ public void uploadFromFile(String filePath, ParallelTransferOptions parallelTran
             throw logger.logExceptionAsError(e);
         }
     }
+
+    /**
+     * Generates a user delegation sas for the blob using the specified
+     * {@link BlobServiceSasSignatureValues}.
+     * @see BlobServiceSasSignatureValues for more information on how to construct a user delegation SAS.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobClient.generateUserDelegationSas#BlobServiceSasSignatureValues-UserDelegationKey}
+     *
+     * @param blobServiceSasSignatureValues {@link BlobServiceSasSignatureValues}
+     * @param userDelegationKey {@link UserDelegationKey}
```

Comment:
>link [](start = 34, length = 4)

This is a generated type and doesn't have great docs on it, so we should put an actual doc string and link to the method that they can use to retrieve one

Reply:
I'll replace this with the following 

     * @param userDelegationKey A {@link UserDelegationKey} object used to sign the SAS values.
     * @see BlobServiceClient#getUserDelegationKey(OffsetDateTime, OffsetDateTime) for more information on how to get a
     * user delegation key.

Reply:
Changed in BlobAsyncClientBase, BlobClientBase, BlobContainerClient, BlobContainerAsyncClient